internal/framework/service/rulesets/schema.go

```diff
@@ -108,10 +108,16 @@ func (r *RulesetResource) Schema(ctx context.Context, req resource.SchemaRequest
 						consts.IDSchemaKey: schema.StringAttribute{
 							Computed:            true,
 							MarkdownDescription: "Unique rule identifier.",
+							PlanModifiers: []planmodifier.String{
+								stringplanmodifier.UseStateForUnknown(),
+							},
 						},
 						"ref": schema.StringAttribute{
-							Optional:            true,
-							Computed:            true,
+							Optional: true,
+							Computed: true,
+							PlanModifiers: []planmodifier.String{
+								stringplanmodifier.UseStateForUnknown(),
```

Comment:
@zakcutner we're pushing folks to use this however, it cannot be defined once the rule is created. should we also mark this as [`RequiresReplace()`](https://developer.hashicorp.com/terraform/plugin/framework/resources/plan-modification#requiresreplace) as well?

Reply:
We cannot do this unfortunately, because it's only the rule (and not the whole ruleset resource) that will be replaced if the `ref` changes

Reply:
does the `UseStateForUnknown()` make sense?